### ISSUE TITLE
`useForm` with boolean `defaultValue` returns incorrectly typed data

### ISSUE THREAD

**Issue by aaronadamsCA** [edmundhung/conform#373]
### Describe the bug and the expected behavior

If I create a schema with a boolean field, and pass in a boolean `defaultValue` for that field, `useForm()` gives me back incorrectly typed data at `field.defaultValue`. The field's type is `string | undefined`, but its value is still a boolean.

This causes further problems: for example, `conform.input(config, { type: "checkbox" })` calculates an incorrect `defaultChecked` value, which is how I discovered this problem.

### Conform version

v0.9.1

### Steps to Reproduce the Bug or Issue

[CodeSandbox link](https://codesandbox.io/p/sandbox/hopeful-andras-f3pwvd?file=%2Fsrc%2FApp.tsx&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clr7y1o3m00lv3p6e4ppt2srk%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clr7y1o3l00ls3p6eky4u27zd%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clr7y1o3l00lu3p6eo0aw3sjq%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clr7y1o3l00lt3p6eeoj29gr9%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clr7y1o3l00ls3p6eky4u27zd%2522%253A%257B%2522id%2522%253A%2522clr7y1o3l00ls3p6eky4u27zd%2522%252C%2522activeTabId%2522%253A%2522clr7y1o5s00m83p6eqpqzn98m%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr7y1jgp00lg3p6e8xkrjion%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522id%2522%253A%2522clr7y1o5s00m83p6eqpqzn98m%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clr7y1o3l00lt3p6eeoj29gr9%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr7wt26p00043p6feftx2z4s%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clr7y1o3l00lt3p6eeoj29gr9%2522%252C%2522activeTabId%2522%253A%2522clr7wt26p00043p6feftx2z4s%2522%257D%252C%2522clr7y1o3l00lu3p6eo0aw3sjq%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clr7y1o3l00lu3p6eo0aw3sjq%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Afalse%252C%2522sidebarPanelSize%2522%253A15%257D)

1. Create a Zod schema with a `boolean` field:

    ```tsx
    const schema = z.object({
      foo: z.boolean().default(false),
    });
    ```

2. Call `useForm` with a default `boolean` value:

    ```tsx
    const [form, { foo }] = useForm({
      defaultValue: {
        foo: true,
      },
      onValidate: ({ formData }) => parse(formData, { schema }),
    });
    ```

**Expected result:** `foo.defaultValue` has a value of `"on"` (probably), since its type is defined as `string | undefined`.

**Actual result:** `foo.defaultValue` has a value of `true`, which does not match its type!

### Additional context

I know there are some known issues in this area, but I didn't see this specific problem reported yet.

**Comment by edmundhung** [comment]
Thanks for reporting. This seems to be a type issue in the current version as it supposes to [enforce the default value to be serialized](https://github.com/edmundhung/conform/blob/6e14734289fc5c54998ae08815f15cb733533469/packages/conform-react/hooks.ts#L62-L72) and so you will never be able to provide a boolean there.

But anyway, this should be resolved in the upcoming v1 as there will be a default serialization done if it finds anything that isn't a `string`, e.g. transforming boolean to `on` as you suggested. I will keep the issue open until v1 is out.

**Comment by aaronadamsCA** [comment]
I can confirm this works great in v1.0.0-rc.0. Thank you!

**Comment by edmundhung** [comment]
This should be resolved in [v1.0.0](https://github.com/edmundhung/conform/releases/tag/v1.0.0).
